### ISSUE TITLE
Exec mixin doesn't notice a file permission change

### ISSUE THREAD

**Issue by martinburchell** [getporter/porter#1818]
## Describe the bug
The exec mixin will complain if a file is not executable. Once it is executable Porter doesn't notice until you rename it.

## To Reproduce
1. Start off with a simple` porter.yaml` that uses the `exec` mixin:
https://github.com/martinburchell/porter-test/tree/file-permission-bug
2. Make sure the `wrong_permission.sh` script is not executable
3. Run `porter install`. The should be a `permission denied` failure
4. Change the `wrong_permission.sh` script to be executable 
5. Run `porter install`. You still get the `permission denied` failure. It only works if you rename the file and change `porter.yaml` to reflect the new name and then revert to the previous name. The `--force` flag or `porter installation delete` have no effect.

## Expected behavior
No `permission denied` failure

## Version
porter v0.38.7 (43a2b2ae)

**Comment by carolynvs** [comment]
Yeah, this is a known gap in our logic to detect if the bundle is out of date when `porter publish` is called. Right now only some changes, such as changing the porter.yaml or the version of porter will cause publish to detect that the bundle is stale and rebuild automatically for you.

A better workflow until this is fixed is to always call porter build first, before calling porter publish.

I thought that there was an existing issue to track this feature request but since I can't find it, we'll track it under this one. 👍
